### DISCUSSION TITLE
add a custom font

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#4433] [Help]
Hello,

I hope you are fine.

I am using a custom font for my client - a company very touchy about its image...

How can I use the 

@font-face {
    font-family: 'YourFontName'; /*a name to be used later*/
    src: url('http://domain.com/fonts/font.ttf'); /*URL to font*/
}

in budibase?

Thanks a lot

**Comment by melohagan** [comment]
Hi @oswaldB 

I did find this old issue: https://github.com/Budibase/budibase/issues/196

It is closed though, so it may simply not be a priority.

You can do inline fonts with custom CSS, e.g. `font-family: "Times New Roman", Times, serif;` 
Might do if there was a standard web font close to what you need.

**Comment by Havock94** [comment]
I found a hack-ish workaround while waiting for a proper implementation of this.
In the Navigation Layout, I added an `Embed` block from `Elements > Embed` and bound to the following Javascript code

```
return window.loadedFont ? '' : window.loadedFont = '<link id="customFont" rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css2?family=Lexend:wght@100;300;500;700;900&display=swap">';
```
_(Change the link to import your actual font)_

This allowed me to have an embedded element that loads the custom font only once (without the `window.loadedFont` part you'd import the font each re-render).

Now we've imported the font but we actually have to apply it to the DOM elements.
So I've selected the `Navigation Layout` from the design tree
![image](https://user-images.githubusercontent.com/7635248/167167383-bba6759e-6254-4eac-9a0a-1fac12037411.png)
and set its `Custom CSS` to

```
font-family: "YOUR_FONT_NAME" !important;
--spectrum-global-font-family-base: "YOUR_FONT_NAME", adobe-clean, 'Source Sans Pro', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Trebuchet MS', 'Lucida Grande', sans-serif !important;
--spectrum-alias-body-text-font-family: "YOUR_FONT_NAME", var(--spectrum-global-font-family-base) !important;
```

This overwrites the inherited font-family of most (if not all) elements in the app, falling back to the original ones in case something isn't working.

Note that any Budibase update that would change the default fonts for elements could break fallback fonts in the script, so again, this is just a **temporary** workaround until an official way is provided.


Hope this helped!

**Reply by melohagan** [reply]
Thank you very much for sharing!

**Reply by Geedeeel** [reply]
Works like a charm, thanks!

**Comment by poirazis** [comment]
Is there a similar approach to be able to inject custom CSS ??

**Comment by Junaida432** [comment]
Yes, you can use a custom font with `@font-face` in Budibase.

You can also convert your font into web-friendly formats such as WOFF/WOFF2, which are generally better for websites and web apps. Upload ttf or otf file and convert which you want type.

I actually built a small tool for this myself, called DafontStyle Font Converter. You can try it here:

https://font-converter.dafontstyle.io/

<img width="1474" height="836" alt="Screenshot 2026-09-08 112618" src="https://github.com/user-attachments/assets/be8f0f06-8037-40f2-a6f4-269012260959" />


I’m not promoting it or anything — I made the tool myself and thought it might be useful for your use case. 🙂
